shm/Ship.hpp

```diff
@@ -2,31 +2,34 @@
 #include <memory>
 #include <string>
 #include <vector>
-#include "Cargo.hpp"
+
+class Cargo;
 
 class Ship {
+public:
+    using CargoVec = std::vector<std::unique_ptr<Cargo>>;
+    Ship();
+    Ship(size_t capacity, size_t maxCrew, size_t speed, const std::string& name, const size_t id, CargoVec cargoVec);
+    Ship(size_t maxCrew, size_t speed, size_t id);
+ 
+    void setName(const std::string& name);
+ 
+    Ship& operator-=(size_t num);
```

Comment:
Don't check if go below 0. 
Parameter num don't give meaning of what we add or substract.
Maybe should change operator for meaningful  addCrew().